### ISSUE TITLE
receving error when running task using mysql plugin

### ISSUE THREAD

**Issue by imvisiontech** [intelsdi-x/snap#894]
Hi,
I am trying to run a task using the meminfo collector and mysql publisher.

./snapctl task create -t /home/shirley/temp/snap/examples/tasks/mem-mysql_1.json
the json file content for reference is pasted below.
the task runs but fails with the following error:
Publish call error: unsupported type
the table created at mysql DB:
+---------------+--------------+------+-----+---------+-------+
| Field         | Type         | Null | Key | Default | Extra |
+---------------+--------------+------+-----+---------+-------+
| timestamp     | varchar(200) | YES  |     | NULL    |       |
| source_column | varchar(200) | YES  |     | NULL    |       |
| key_column    | varchar(200) | YES  |     | NULL    |       |
| value_column  | varchar(200) | YES  |     | NULL    |       |
+---------------+--------------+------+-----+---------+-------+

what can be the root cause for that problem?
where can I find reference for understanding the errors received?

thanks,
Shirley.

{
    "version": 1,
    "schedule": {
        "type": "simple",
        "interval": "1s"
    },
    "workflow": {
        "collect": {
            "metrics": {
                "/intel/procfs/meminfo/MemFree": {},
                "/intel/procfs/meminfo/MemAvailable": {},
                "/intel/procfs/meminfo/MemTotal": {},
                "/intel/procfs/meminfo/MemUsed": {}
            },
            "process": [
                {
                    "plugin_name": "passthru",
                    "process": null,
                    "publish": [
                        {
                            "plugin_name": "mysql",
                            "config": {
                                "hostname": "10.0.0.203",
                                "port": 3306,
                                "username": "root",
                                "tablename": "snap",
                                "database": "imVisionSsoDB",
                                "password": "123456"
                            }
                        }
                    ],
                    "config": null
                }
            ],
            "publish": null
        }
    }
}

**Comment by imvisiontech** [comment]
I also tried it with the postgresql and received the same error.

**Comment by IRCody** [comment]
>Publish call error: unsupported type

This is saying there as an error when calling PublishMetrics and that error was 'unsupported type'. Looking into it seems like the error it's hitting is [here](https://github.com/intelsdi-x/snap-plugin-publisher-mysql/blob/master/mysql/mysql.go#L190). I think this is because meminfo seems to publish it's metrics as uints while mysql has a method it calls on all data coming in to convert it to a string that does not deal with uints.

It seems like postgresql has a similar issue. I opened issues on those plugins for support uint's. Meminfo is not the only plugin that emits uints.

**Comment by mjbrender** [comment]
Hey @IRCody - tagging this as a question about the Snap Framework as opposed to a bug for it. Do you agree?

**Comment by IRCody** [comment]
@mjbrender: I agree it's really a plugin issue. The framework is accurately reporting the bug reported to it from the plugin.

**Comment by mjbrender** [comment]
Still checking into this @imvisiontech - thanks for bringing it up! cc @intelsdi-x/plugin-maintainers @intelsdi-x/snap-maintainers
